File: CareConnect/CareConnect.Models/Responses/WorkshopType.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CareConnect.Models.Responses
{
    public class WorkshopType
    {
        public int WorkshopTypeId { get; set; }

        public string Name { get; set; } = null!;

        public string? Description { get; set; }

        public DateTime ModifiedDate { get; set; }
     }
}

File: UI/careconnect_admin/lib/models/responses/workshop_type.dart
import 'package:json_annotation/json_annotation.dart';

part 'workshop_type.g.dart'; // Don't open or edit this

@JsonSerializable()
class WorkshopType {
  final int workshopTypeId;
  final String name;
  final String? description;
  final DateTime modifiedDate;

  WorkshopType({
    required this.workshopTypeId,
    required this.name,
    this.description,
    required this.modifiedDate,
  });

  factory WorkshopType.fromJson(Map<String, dynamic> json) =>
      _$WorkshopTypeFromJson(json);

  Map<String, dynamic> toJson() => _$WorkshopTypeToJson(this);
}
